rust/datafusion/src/physical_plan/expressions.rs

```diff
@@ -97,766 +104,712 @@ pub fn col(name: &str) -> Arc<dyn PhysicalExpr> {
 /// SUM aggregate expression
 #[derive(Debug)]
 pub struct Sum {
+    name: String,
```

Comment:
`AggregateExpr` has this information with them now because it allows them to `create_accumulator` without access to the input schema. This is helpful because, on the second pass, we need to create accumulators on the fly, and the input_schema of the second pass is different, as it now corresponds to the schema 

`[group1, group2, agg1_state1, agg1_state2, ...]`